src/workers/_reader_h5.js

```diff
@@ -0,0 +1,120 @@
+import * as scran from "scran.js";
+import * as utils from "./_utils.js";
+import * as rutils from "./_reader_utils.js";
+
+
+export class H5Reader {
+    constructor(files) {
+        this.files = files;
+
+        this.cache = {};
+        this.parameters = {};
+        this.abbreviated = {};
```

Comment:
Why do these things need to be part of the class? It doesn't really seem like we need that kind of statefulness given that we use each component immediately in `_inputs.compute()` and `_preflight.compute()`. Rather, it should be possible for each method to directly return the pieces of interest.